src/arduino/arduinoSettings.ts

```diff
@@ -56,7 +56,11 @@ export class ArduinoSettings implements IArduinoSettings {
 
     public get defaultExamplePath(): string {
         if (os.platform() === "darwin") {
-            return path.join(this._arduinoPath, "Arduino.app/Contents/Java/examples");
+            if (this._arduinoPath.match(/Arduino.*\.app/)){

```

Comment:
5 Duplicate if logics. 
Suggest to add a utility method resolveMacArduinoAppPath in src/common/util.ts. Then just need join "/Contents/xxx"

resolveMacArduinoAppPath(arduinoPath: string): string {
    if (/Arduino.*\.app/.test(arduinoPath)) {
            return arduinoPath;
    } else {
            return path.join(arduinoPath, "Arduino.app");
    }
}